### PR TITLE
Select from system tables when table based on table function.

### PR DESCRIPTION
**Pull request by MikhailBurdukov**
### Changelog category (leave one):
- Bug Fix (user-visible misbehavior in an official stable release)

### Changelog entry (a user-readable short description of the changes that goes to CHANGELOG.md):

Prevent reference to a remote data source for the `data_paths` column in `system.tables` if the table is created with a table function using explicit column description.

Example:
```
CREATE table test_pg (id Int, value Int) AS postgresql('localhost:5432', 'test_db', 'test_table', 'test1', '123456')

SELECT data_paths FROM system.tables WHERE name='test_pg'
```
Result:
```
┌─data_paths─┐
│ []         │
└────────────┘
```
### Documentation entry for user-facing changes

- [ ] Documentation is written (mandatory for new features)

> Information about CI checks: https://clickhouse.com/docs/en/development/continuous-integration/

### CODE REVIEW

**Comment by robot-clickhouse** [comment]
*This is an automated comment for commit 15956a690dd4d2217e8dde383fc40a308630ba04 with description of existing statuses. It's updated for the latest CI running*

[⏳ Click here](https://s3.amazonaws.com/clickhouse-test-reports/55540/15956a690dd4d2217e8dde383fc40a308630ba04/ci_running.html) to open a full report in a separate page

<details><summary>Successful checks</summary>
<table>
<thead><tr><th>Check name</th><th>Description</th><th>Status</th></tr></thead>
<tbody>
<tr><td>Mergeable Check</td><td>Checks if all other necessary checks are successful</td><td><a href="https://github.com/ClickHouse/ClickHouse/actions/runs/6493193568">✅ success</a></td></tr>
<tbody>
</table>
</details>
<table>
<thead><tr><th>Check name</th><th>Description</th><th>Status</th></tr></thead>
<tbody>
<tr><td>CI running</td><td>A meta-check that indicates the running CI. Normally, it's in <b>success</b> or <b>pending</b> state. The failed status indicates some problems with the PR</td><td><a href="https://s3.amazonaws.com/clickhouse-test-reports/55540/15956a690dd4d2217e8dde383fc40a308630ba04/ci_running.html">⏳ pending</a></td></tr>
<tbody>
</table>

File: src/Storages/StorageProxy.h [hunk 1]
```diff
@@ -152,7 +152,7 @@ class StorageProxy : public IStorage
     CheckResults checkData(const ASTPtr & query, ContextPtr context) override { return getNested()->checkData(query, context); }
     void checkTableCanBeDropped([[ maybe_unused ]] ContextPtr query_context) const override { getNested()->checkTableCanBeDropped(query_context); }
     bool storesDataOnDisk() const override { return getNested()->storesDataOnDisk(); }
-    Strings getDataPaths() const override { return getNested()->getDataPaths(); }
+    Strings getDataPaths() const override { return {}; }
```
Comment (tavplubix) [inline src/Storages/StorageProxy.h:155]: But it will return wrong result

Comment (MikhailBurdukov) [reply src/Storages/StorageProxy.h:155]: As I understood `StorageProxy` keeps tables based on table functions and  with `StorageMaterializedMySQL` engine. Seems like the second one may have its own data. Can we always return empty value for `StorageTableFunctionProxy` class?

Comment (tavplubix) [reply src/Storages/StorageProxy.h:155]: I'm not sure, see, for example, table function `file`

Comment (MikhailBurdukov) [reply src/Storages/StorageProxy.h:155]: `file` function have not empty value:``

**Comment by antaljanosbenjamin** [comment]
I am trying to understand the change, but the test succeeds on my machine even without your changes. The `data_paths` fields is an empty array.

I am not super familiar with the implementation of postgres/mysql table functions, that's why I was happy to review this PR, so I can learn.

As far as I can see in case of PostgreSQL/MySQL, the `StorageProxy::getNested()` will return `StoragePostgreSQL`/`StorageMySQL`. As none of them override the default implementation of `getDataPaths`, in `IStorage`, I don't see how the `data_paths` would contain anything but an empty array.

The sources I used to collect the above informations:
1. `ITableFunction.cpp`: the [implementation](https://github.com/ClickHouse/ClickHouse/blob/8a1363bcf1124498fd46f68e2aa671e5175bf321/src/TableFunctions/ITableFunction.cpp#L43-L46) of `get_storage` just call `executeImpl` of the table function.
2. `TableFunctionPostgreSQL`: the [implementation](https://github.com/ClickHouse/ClickHouse/blob/8a1363bcf1124498fd46f68e2aa671e5175bf321/src/TableFunctions/TableFunctionPostgreSQL.cpp#L22-L38) of `executeImpl` returns a `StoragePostgreSQL` object
3. `StoragePostgreSQL.h`: it [doesn't override](https://github.com/ClickHouse/ClickHouse/blob/master/src/Storages/StoragePostgreSQL.h) `getDataPaths()`

Do I miss something?

**Comment by MikhailBurdukov** [comment]
The motivation of this PR is to make selecting the `data_paths` column without throwing the exception is cases when source of the data is table function.

>but the test succeeds on my machine even without your changes

This is strange. Have tested with  build from the master branch:
```
git log 
commit ebfad5c8b886aafbea656e7fbecff277b6e801fe (origin/master, origin/HEAD, master)
Merge: e64abfc557e db1b9408f39
Author: Alexander Tokmakov <tavplubix@clickhouse.com>
Date:   Fri Oct 13 13:07:28 2023 +0200

    Merge pull request #55587 from azat/fix-replica-group
    
    Fix replica groups for Replicated database engine

commit db1b9408f398634eddbf7c50418872805ee57dd3
Author: Azat Khuzhin <a3at.mail@gmail.com>
Date:   Fri Oct 13 12:25:33 2023 +0200

    Fix replica groups for Replicated database engine
    
    Should fix 02447_drop_database_replica test
    
    Signed-off-by: Azat Khuzhin <a.khuzhin@semrush.com>

commit e64abfc557e7595a7801eae29ccb12153bc7f7f4
Merge: 8a1363bcf11 3aac663bd26
Author: vdimir <vdimir@clickhouse.com>
Date:   Fri Oct 13 12:29:58 2023 +0200

    Merge pull request #49555 from ClickHouse/vdimir/analyzer_fix_join_tests
    
    Fixing join tests with analyzer
```
Also with the latest ClickHouse build from `curl https://clickhouse.com/ | sh`
The result is the same:
```
PATH=/data/disk2/home/mburdukov/workspace/clickhouse_test:$PATH ./tests/clickhouse-test 02888_system_tables_with_inaccsessible_table_function
Using queries from '/data/disk2/home/mburdukov/workspace/clickhouse_src/tests/queries' directory
Connecting to ClickHouse server... OK

Running 1 stateless tests (MainProcess).

02888_system_tables_with_inaccsessible_table_function:                  [ FAIL ] - return code: 102
[mburdukov] 2023.10.17 10:00:22.760461 [ 2904597 ] {306d1b9b-a8b4-4756-8aad-a992738e127f} <Error> PostgreSQLConnectionPool: Connection error: connection to server at "127.121.0.1", port 5432 failed: Connection refused
        Is the server running on that host and accepting TCP/IP connections?

[mburdukov] 2023.10.17 10:00:22.769409 [ 2904597 ] {306d1b9b-a8b4-4756-8aad-a992738e127f} <Error> PostgreSQLConnectionPool: Connection error: connection to server at "127.121.0.1", port 5432 failed: Connection refused
        Is the server running on that host and accepting TCP/IP connections?

[mburdukov] 2023.10.17 10:00:22.769557 [ 2904597 ] {306d1b9b-a8b4-4756-8aad-a992738e127f} <Error> PostgreSQLConnectionPool: Connection error: connection to server at "127.121.0.1", port 5432 failed: Connection refused
        Is the server running on that host and accepting TCP/IP connections?

[mburdukov] 2023.10.17 10:00:22.769662 [ 2904597 ] {306d1b9b-a8b4-4756-8aad-a992738e127f} <Error> PostgreSQLConnectionPool: Connection error: connection to server at "127.121.0.1", port 5432 failed: Connection refused
        Is the server running on that host and accepting TCP/IP connections?

[mburdukov] 2023.10.17 10:00:22.769758 [ 2904597 ] {306d1b9b-a8b4-4756-8aad-a992738e127f} <Error> PostgreSQLConnectionPool: Connection error: connection to server at "127.121.0.1", port 5432 failed: Connection refused
        Is the server running on that host and accepting TCP/IP connections?

[mburdukov] 2023.10.17 10:00:22.776743 [ 2904205 ] {306d1b9b-a8b4-4756-8aad-a992738e127f} <Error> executeQuery: Code: 614. DB::Exception: Try 5. Connection to `127.121.0.1:5432` failed with error: connection to server at "127.121.0.1", port 5432 failed: Connection refused
        Is the server running on that host and accepting TCP/IP connections?

: While executing Tables. (POSTGRESQL_CONNECTION_FAILURE) (version 23.10.1.1018 (official build)) (from [::1]:33352) (comment: 02888_system_tables_with_inaccsessible_table_function.sql) (in query: SELECT name, engine, engine_full, create_table_query, data_paths, notEmpty([metadata_path]), notEmpty([uuid]) FROM system.tables WHERE name like '%tablefunc%' and database=currentDatabase() ORDER BY metadata_modification_time;), Stack trace (when copying this message, always include the lines below):

0. ./build_docker/./src/Common/Exception.cpp:103: DB::Exception::Exception(DB::Exception::MessageMasked&&, int, bool) @ 0x000000000c7944b7 in /data/disk2/home/mburdukov/workspace/clickhouse_test/clickhouse
1. ./build_docker/./contrib/llvm-project/libcxx/include/string:1499: DB::Exception::Exception(PreformattedMessage const&, int) @ 0x0000000010b1dacc in /data/disk2/home/mburdukov/workspace/clickhouse_test/clickhouse
2. ./build_docker/./src/Core/PostgreSQL/PoolWithFailover.cpp:135: postgres::PoolWithFailover::get() @ 0x0000000010efda4b in /data/disk2/home/mburdukov/workspace/clickhouse_test/clickhouse
3. ./build_docker/./src/Storages/StoragePostgreSQL.cpp:98: DB::StoragePostgreSQL::getTableStructureFromData(std::shared_ptr<postgres::PoolWithFailover> const&, String const&, String const&, std::shared_ptr<DB::Context const> const&) @ 0x000000001294d907 in /data/disk2/home/mburdukov/workspace/clickhouse_test/clickhouse
4. ./build_docker/./src/Storages/StoragePostgreSQL.cpp:0: DB::StoragePostgreSQL::StoragePostgreSQL(DB::StorageID const&, std::shared_ptr<postgres::PoolWithFailover>, String const&, DB::ColumnsDescription const&, DB::ConstraintsDescription const&, String const&, std::shared_ptr<DB::Context const>, String const&, String const&) @ 0x000000001294d3e0 in /data/disk2/home/mburdukov/workspace/clickhouse_test/clickhouse
5. ./build_docker/./contrib/llvm-project/libcxx/include/__memory/shared_ptr.h:1460: std::shared_ptr<DB::StoragePostgreSQL> std::allocate_shared[abi:v15000]<DB::StoragePostgreSQL, std::allocator<DB::StoragePostgreSQL>, DB::StorageID, std::shared_ptr<postgres::PoolWithFailover> const&, String const&, DB::ColumnsDescription, DB::ConstraintsDescription, String, std::shared_ptr<DB::Context const>&, String const&, String const&, void>(std::allocator<DB::StoragePostgreSQL> const&, DB::StorageID&&, std::shared_ptr<postgres::PoolWithFailover> const&, String const&, DB::ColumnsDescription&&, DB::ConstraintsDescription&&, String&&, std::shared_ptr<DB::Context const>&, String const&, String const&) @ 0x0000000010b6ff5a in /data/disk2/home/mburdukov/workspace/clickhouse_test/clickhouse
6. ./build_docker/./contrib/llvm-project/libcxx/include/__memory/shared_ptr.h:962: DB::TableFunctionPostgreSQL::executeImpl(std::shared_ptr<DB::IAST> const&, std::shared_ptr<DB::Context const>, String const&, DB::ColumnsDescription, bool) const @ 0x0000000010b6ee81 in /data/disk2/home/mburdukov/workspace/clickhouse_test/clickhouse
7. ./build_docker/./src/TableFunctions/ITableFunction.cpp:45: std::shared_ptr<DB::IStorage> std::__function::__policy_invoker<std::shared_ptr<DB::IStorage> ()>::__call_impl<std::__function::__default_alloc_func<DB::ITableFunction::execute(std::shared_ptr<DB::IAST> const&, std::shared_ptr<DB::Context const>, String const&, DB::ColumnsDescription, bool, bool) const::$_0, std::shared_ptr<DB::IStorage> ()>>(std::__function::__policy_storage const*) @ 0x0000000010dc8087 in /data/disk2/home/mburdukov/workspace/clickhouse_test/clickhouse
8. ./build_docker/./src/Storages/StorageTableFunction.h:44: DB::StorageTableFunctionProxy::getNestedImpl() const @ 0x0000000010dcbdf0 in /data/disk2/home/mburdukov/workspace/clickhouse_test/clickhouse
9. ./build_docker/./src/Storages/StorageProxy.h:155: DB::StorageProxy::getDataPaths() const @ 0x0000000010dcb798 in /data/disk2/home/mburdukov/workspace/clickhouse_test/clickhouse
10. ./build_docker/./contrib/llvm-project/libcxx/include/vector:537: DB::TablesBlockSource::generate() @ 0x0000000010beac82 in /data/disk2/home/mburdukov/workspace/clickhouse_test/clickhouse
11. ./build_docker/./src/Processors/Chunk.h:90: DB::ISource::tryGenerate() @ 0x00000000133a9fd8 in /data/disk2/home/mburdukov/workspace/clickhouse_test/clickhouse
12. ./build_docker/./contrib/llvm-project/libcxx/include/optional:344: DB::ISource::work() @ 0x00000000133a9b0a in /data/disk2/home/mburdukov/workspace/clickhouse_test/clickhouse
13. ./build_docker/./src/Processors/Executors/ExecutionThreadContext.cpp:0: DB::ExecutionThreadContext::executeTask() @ 0x00000000133c19da in /data/disk2/home/mburdukov/workspace/clickhouse_test/clickhouse
14. ./build_docker/./src/Processors/Executors/PipelineExecutor.cpp:273: DB::PipelineExecutor::executeStepImpl(unsigned long, std::atomic<bool>*) @ 0x00000000133b8490 in /data/disk2/home/mburdukov/workspace/clickhouse_test/clickhouse
15. ./build_docker/./contrib/llvm-project/libcxx/include/__memory/shared_ptr.h:701: DB::PipelineExecutor::execute(unsigned long, bool) @ 0x00000000133b7720 in /data/disk2/home/mburdukov/workspace/clickhouse_test/clickhouse
16. ./build_docker/./src/Processors/Executors/PullingAsyncPipelineExecutor.cpp:0: void std::__function::__policy_invoker<void ()>::__call_impl<std::__function::__default_alloc_func<ThreadFromGlobalPoolImpl<true>::ThreadFromGlobalPoolImpl<DB::PullingAsyncPipelineExecutor::pull(DB::Chunk&, unsigned long)::$_0>(DB::PullingAsyncPipelineExecutor::pull(DB::Chunk&, unsigned long)::$_0&&)::'lambda'(), void ()>>(std::__function::__policy_storage const*) @ 0x00000000133c53ef in /data/disk2/home/mburdukov/workspace/clickhouse_test/clickhouse
17. ./build_docker/./base/base/../base/wide_integer_impl.h:809: void* std::__thread_proxy[abi:v15000]<std::tuple<std::unique_ptr<std::__thread_struct, std::default_delete<std::__thread_struct>>, void ThreadPoolImpl<std::thread>::scheduleImpl<void>(std::function<void ()>, Priority, std::optional<unsigned long>, bool)::'lambda0'()>>(void*) @ 0x000000000c87df67 in /data/disk2/home/mburdukov/workspace/clickhouse_test/clickhouse
18. ? @ 0x00007ff7fb843ac3 in ?
19. ? @ 0x00007ff7fb8d5a40 in ?

Received exception from server (version 23.10.1):
Code: 614. DB::Exception: Received from localhost:9000. DB::Exception: Try 5. Connection to `127.121.0.1:5432` failed with error: connection to server at "127.121.0.1", port 5432 failed: Connection refused
        Is the server running on that host and accepting TCP/IP connections?

: While executing Tables. (POSTGRESQL_CONNECTION_FAILURE)
(query: SELECT name, engine, engine_full, create_table_query, data_paths, notEmpty([metadata_path]), notEmpty([uuid])
    FROM system.tables
    WHERE name like '%tablefunc%' and database=currentDatabase()
    ORDER BY metadata_modification_time;)
, result:



stdout:

```

>As none of them override the default implementation of getDataPaths

Yes. The problem is to get the `data_paths` field `StoragePostgeSQL/StorageMySQL` class is  required. If in the `Create ... AS postgresql/mysql(...)` statement [`schema`](https://clickhouse.com/docs/en/sql-reference/table-functions/postgresql) is not  specified then we will try to get it from the source in the class constructor( [Here for example](https://github.com/ClickHouse/ClickHouse/blob/914e6020c0214a5e7aa5b1c136e556524d1ca538/src/Storages/StoragePostgreSQL.cpp#L78) ). As a result, we will get an exception because the source of the table is unreachable.

**Comment by antaljanosbenjamin** [comment]
> This is strange. Have tested with build from the master branch:

Okay, based on your example I assume you mean when the data source is not available, right? Because in that case I noticed that issue. It wasn't clear for me from the description of the PR. Thanks for the clarification.

I will try to think about this, but cannot promise anything.

**Comment by MikhailBurdukov** [comment]
@antaljanosbenjamin Hi! Could you review the PR? Thx in advance!

File: tests/queries/0_stateless/02888_system_tables_with_inaccsessible_table_function.sql [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+    WHERE name like '%tablefunc%' and database=currentDatabase()
+    ORDER BY name;
+
+DETACH TABLE {CLICKHOUSE_DATABASE:Identifier}.tablefunc01; 
+DETACH TABLE {CLICKHOUSE_DATABASE:Identifier}.tablefunc02;
+
+
+ATTACH TABLE {CLICKHOUSE_DATABASE:Identifier}.tablefunc01;
+ATTACH TABLE {CLICKHOUSE_DATABASE:Identifier}.tablefunc02;
+
```
Comment (antaljanosbenjamin) [inline tests/queries/0_stateless/02888_system_tables_with_inaccsessible_table_function.sql:28]:
```suggestion
DETACH TABLE {CLICKHOUSE_DATABASE:Identifier}.tablefunc02;
DETACH TABLE {CLICKHOUSE_DATABASE:Identifier}.tablefunc03;


ATTACH TABLE {CLICKHOUSE_DATABASE:Identifier}.tablefunc01;
ATTACH TABLE {CLICKHOUSE_DATABASE:Identifier}.tablefunc02;
ATTACH TABLE {CLICKHOUSE_DATABASE:Identifier}.tablefunc03;

```

File: tests/queries/0_stateless/02888_system_tables_with_inaccsessible_table_function.sql [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+
+CREATE TABLE {CLICKHOUSE_DATABASE:Identifier}.tablefunc01 (x int) AS postgresql('127.121.0.1:5432', 'postgres_db', 'postgres_table', 'postgres_user', '124444');
+CREATE TABLE {CLICKHOUSE_DATABASE:Identifier}.tablefunc02 (x int) AS mysql('127.123.0.1:3306', 'mysql_db', 'mysql_table', 'mysql_user','123123');
+CREATE TABLE {CLICKHOUSE_DATABASE:Identifier}.tablefunc03 (a int) As sqlite('db_path', 'table_name');
+CREATE TABLE {CLICKHOUSE_DATABASE:Identifier}.tablefunc04 (a int) AS  mongodb('127.0.0.1:27017','test', 'my_collection', 'test_user', 'password', 'a Int');
+CREATE TABLE {CLICKHOUSE_DATABASE:Identifier}.tablefunc05 (a int) AS redis('127.0.0.1:6379', 'key', 'key UInt32');
+
+CREATE TABLE {CLICKHOUSE_DATABASE:Identifier}.tablefunc01_without_schema  AS postgresql('127.121.0.1:5432', 'postgres_db', 'postgres_table', 'postgres_user', '124444'); -- { serverError 614 }
+CREATE TABLE {CLICKHOUSE_DATABASE:Identifier}.tablefunc02_without_schema  AS mysql('127.123.0.1:3306', 'mysql_db', 'mysql_table', 'mysql_user','123123'); -- {serverError 279 }
```
Comment (antaljanosbenjamin) [inline tests/queries/0_stateless/02888_system_tables_with_inaccsessible_table_function.sql:15]:
```suggestion
CREATE TABLE {CLICKHOUSE_DATABASE:Identifier}.tablefunc04 (a int) AS mongodb('127.0.0.1:27017','test', 'my_collection', 'test_user', 'password', 'a Int');
CREATE TABLE {CLICKHOUSE_DATABASE:Identifier}.tablefunc05 (a int) AS redis('127.0.0.1:6379', 'key', 'key UInt32');

CREATE TABLE {CLICKHOUSE_DATABASE:Identifier}.tablefunc01_without_schema AS postgresql('127.121.0.1:5432', 'postgres_db', 'postgres_table', 'postgres_user', '124444'); -- { serverError 614 }
CREATE TABLE {CLICKHOUSE_DATABASE:Identifier}.tablefunc02_without_schema AS mysql('127.123.0.1:3306', 'mysql_db', 'mysql_table', 'mysql_user','123123'); -- {serverError 279 }
```

Just tiny cosmetic changes, only necessary if you modify anything, otherwise it doesn't worth to run the CI again just because of these.

File: tests/queries/0_stateless/02888_system_tables_with_inaccsessible_table_function.sql (hunk 2 above)
Comment (antaljanosbenjamin) [reply tests/queries/0_stateless/02888_system_tables_with_inaccsessible_table_function.sql:28]: As you modified also `TableFunctionSQLite`, it is good to check that also in my opinion. It shouldn't hurt the test but the extra checks can find some errors (if there is any).

**Review by antaljanosbenjamin — COMMENTED** [review]
Sorry, I missed your last commit. Thanks for pinging.

**Comment by antaljanosbenjamin** [comment]
Maybe one more thing. The changelog item is not the best in my opinion:
> if the table is created using `AS table_func'.
I think the important part is the presence of schema when using the table functions. I would rephrase that to something like this:
> if the table is created with a table function using explicit column description.

File: src/Storages/StorageTableFunction.h [hunk 4]
```diff
@@ -66,9 +66,7 @@ class StorageTableFunctionProxy final : public StorageProxy
     String getName() const override
     {
         std::lock_guard lock{nested_mutex};
-        if (nested)
-            return nested->getName();
-        return StorageProxy::getName();
+        return getNested()->getName();
```
[the remark this answers is not in the document]
Comment (MikhailBurdukov) [reply src/Storages/StorageTableFunction.h:69]: > What is the reasoning behind it?

Without these changes, the first selection  of the `engine` row from system.tables will return `StorageProxy`, instead of the actual engine name, because nested class does not created. And also does not matter remote source accessible or not.
Example:
```
CREATE table test_pg (id Int, value Int) AS postgresql('localhost:5432', 'test_db', 'test_table', 'test1', '123456')
SELECT engine FROM system.tables WHERE name='test_pg'\G

Row 1:
──────
engine: StorageProxy
```

[the remark this answers is not in the document]
Comment (MikhailBurdukov) [reply src/Storages/StorageTableFunction.h:69]: >Second, if it stays as is, you don't need to lock the nested_mutex. 

Yeh, will fix this

File: tests/queries/0_stateless/02888_system_tables_with_inaccsessible_table_function.sql [hunk 5]
```diff
@@ -0,0 +1,34 @@
+-- Tags: no-fasttest
```
[the remark this answers is not in the document]
Comment (MikhailBurdukov) [reply tests/queries/0_stateless/02888_system_tables_with_inaccsessible_table_function.sql:1]: As I understood table function disabled in the Fasttest. In the first commit CI failed because of:
```
 executeQuery: Code: 46. DB::Exception: Unknown table function postgresql. (UNKNOWN_FUNCTION)
```
https://s3.amazonaws.com/clickhouse-test-reports/55540/15956a690dd4d2217e8dde383fc40a308630ba04/fast_test.html

File: src/Storages/StorageTableFunction.h (hunk 4 above)
[the remark this answers is not in the document]
Comment (tavplubix) [reply src/Storages/StorageTableFunction.h:68]: AFAIR it was intentionally done like this to avoid constructing a nested storage just to get its type. mostly because we may fail to construct it because the remote source may be unavailable

[the remark this answers is not in the document]
Comment (MikhailBurdukov) [reply src/Storages/StorageTableFunction.h]: Reverted it in the last commit

**Review by tavplubix — APPROVED** [review]

**Comment by Alex-Burmak** [comment]
@tavplubix, propose to back port the fix to supported ClickHouse versions. Thanks in advance!
